### ISSUE TITLE
Android platform build error

### ISSUE THREAD

**Issue by folk113** [cisco/openh264#1940]
when i compiled openh264 in android platform ,it display the next problem.
err detail:
in function WelsCommon::PixelAvg_neon(unsigned char*, int, unsigned char const*, int, unsigned char const*, int, int, int)::kpfFuncs:mc.cpp(.data.rel.ro+0x40): error: undefined reference to 'PixStrideAvgWidthEq8_neon'
there is so many error method is end with _neon that i would not paste them;
Solution:
if i edit the Makefile' content{ BUILDTYPE=Debug}，then it will be ok,but i want to complied the library with asm

my build_android.sh file's content
export PATH=~/software/android/sdk/tools:$PATH
make OS=android NDKROOT=~/software/android/ndk-r9d TARGET=android-20

could some one help me? thanks!

**Comment by mstorsjo** [comment]
I can't reproduce this issue - have you tried a clean rebuild?

**Comment by folk113** [comment]
sorry, may be i explain it clearly?
1、i successfully build the library in release mode in default setting;
2、when i extraction the library into a new one, there is no problem when  i have called like WelsCreateSVCEncoder etc, but the method end with _neon。
if i modify the Makefile's content USE_ASM = NO, it will be ok.
# Configurations
ifeq ($(BUILDTYPE), Release)
CFLAGS += $(CFLAGS_OPT)
USE_ASM = No
else
CFLAGS += $(CFLAGS_DEBUG)
USE_ASM = No
endif

3、is this project support arm asm?

我的英文实在太差，下面再用中文描述一下：
1、使用默认的Makefile文件编译，也能成功编出来库；
     下面是我自己调用的时候，再次编译的会出现很多PixStrideAvgWidthEq8_neon类似这样以neon结尾的ln错误。
     我查了一下这些方法，都是存在.S文件中的，一步步跟踪这些方法的编译过程后，发现只有Makefile文件中的USE_ASM＝YES时候才会编译这些方法，也就是上面我修改的地方，然后修改USE_ASM＝NO，再编译一份库，调用的时候不会再出现那些_neon结尾的ln错误，程序能成功运行了。  
2、我看过其它地方提的问题，在2014年上半年有一个回复好像说是不支持汇编，那现在是是因为不支持汇编造成的呢？
3、另外一个问题，我编译出来的android encoder（path:  openh264/codec/build/android/enc）例子也有问题，会提示DoEncoderTest的方法找不到，但是调试时候库已经成功加载了，不知道为何。

**Comment by sijchen** [comment]
看起来是你自己编译的时候没有正确地设置一些汇编库, USE_ASM=No是把汇编代码关掉的意思, 所以在那种情况下你没有问题.

**Comment by folk113** [comment]
please help me to edit my script files  to ensure the asm files loading normal,thanks,
the script files content followed:
{
export PATH=~/software/android/sdk/tools:$PATH
make OS=android NDKROOT=~/software/android/ndk-r9d TARGET=android-20
#修改Makefile文件中的USE_ASM = No，目前版本好像不支持使用汇编
}

**Comment by sijchen** [comment]
sorry I have no idea how to get it right on your environment

**Comment by folk113** [comment]
oh,let it go,the current c version is ok,thanks

**Comment by magiclen** [comment]
I have the same problem while compiling OpenH264 with FFmpeg on Android ARM**64** and Android x86_**64**.

Firstly, I compiled `libopenh264.a` by using

    make install-static OS=android TARGET=android-21 NDKROOT= .....(omit)

For the arch arm64, I also needed to add ASN_ARCH=arm64, or the neon programs could not compile correctly.

By using this way, after compiling, I got two files named libopenh264.a for Android ARM**64** and Android x86_**64**. But when I used each of them in FFmpeg, there were many **undefined reference to XXX_neon** errors at the stage of FFmpeg configuring.

In the end, I set USE_ASM to no and everything was okay.

I don't know whether the performance is bad or not without compiling with ASM.
